src/interceptors/ClientRequest/utils/createRequest.ts

```diff
@@ -25,7 +25,7 @@ export function createRequest(clientRequest: NodeClientRequest): Request {
    * the request "Authorization" header.
    * @see https://github.com/mswjs/interceptors/issues/438
    */
-  if (clientRequest.url.username && clientRequest.url.password) {
+  if (clientRequest.url.username) {
     const auth = `${clientRequest.url.username}:${clientRequest.url.password}`
```

Comment:
Let's then have a fallback for `clientRequest.url.password` to an empty string. If we don't, when the password is not provided this `auth` string will result in 

```
${username}:undefined
```

And produce a malformed encoded auth string.